File: tests/mybatis-python/src/test/python/example/micronaut/CustomConfigurationCustomizer.py
# tag::imports[]
from jakarta.inject import Named, Singleton
from micronaut.configuration.mybatis import MyBatisConfigurationCustomizer
from org.apache.ibatis.session import Configuration
# end::imports[]


# tag::clazz[]
@Named("default")
@Singleton
class CustomConfigurationCustomizer(MyBatisConfigurationCustomizer):
    def customize(self, configuration: Configuration) -> None:
        configuration.addMappers("example.micronaut.mappers")
        configuration.setMapUnderscoreToCamelCase(True)
# end::clazz[]

File: tests/mybatis-java/src/test/java/example/micronaut/CustomConfigurationCustomizer.java
package example.micronaut;

// tag::imports[]
import io.micronaut.configuration.mybatis.MyBatisConfigurationCustomizer;
import io.micronaut.configuration.mybatis.MyBatisMapperScan;
import jakarta.inject.Named;
import jakarta.inject.Singleton;
import org.apache.ibatis.session.Configuration;
// end::imports[]

// tag::clazz[]
@MyBatisMapperScan("example.micronaut.mappers")
@Named("default")
@Singleton
public class CustomConfigurationCustomizer implements MyBatisConfigurationCustomizer {
    @Override
    public void customize(Configuration configuration) {
        configuration.setMapUnderscoreToCamelCase(true);
    }
}
// end::clazz[]
